lib/net/smtp.rb

```diff
@@ -581,6 +581,7 @@ def tlsconnect(s)
       s = ssl_socket(s, @ssl_context)
       logging "TLS connection started"
       s.sync_close = true
+      s.hostname = @address if s.respond_to? :hostname=
```

Comment:
hostname= has to be an ip_address, not a domain name

this example won't work anymore
```ruby
default:
  email_delivery:
    delivery_method: :smtp
    smtp_settings:
      address: smtp.gmail.com
      port: 465
      ssl: true
      authentication: :login
      enable_starttls_auto: true
      user_name: 
      password: 
```

```
An error occurred while sending mail (SSL_connect SYSCALL returned=5 errno=0 state=SSLv2/v3 read server hello A)
```

Reply:
ok, I was wrong, it should be a hostname. Any idea what could cause the error? It works fine without the patch.

Reply:
a pry inside the code was the reason, it actually works fine, thanks for sharing this patch.